raft/src/main/java/org/apache/kafka/raft/MetadataLogConfig.java

```diff
@@ -80,72 +78,55 @@ public class MetadataLogConfig {
-     * @param logSegmentMillis The maximum time before a new metadata log file is rolled out
-     * @param retentionMaxBytes The size of the metadata log and snapshots before deleting old snapshots and log files
-     * @param retentionMillis The time to keep a metadata log file or snapshot before deleting it
-     * @param maxBatchSizeInBytes The largest record batch size allowed in the metadata log
-     * @param maxFetchSizeInBytes The maximum number of bytes to read when fetching from the metadata log
-     * @param deleteDelayMillis The amount of time to wait before deleting a file from the filesystem
-     */
-    public MetadataLogConfig(int logSegmentBytes,
-                             int logSegmentMinBytes,
-                             long logSegmentMillis,
-                             long retentionMaxBytes,
-                             long retentionMillis,
-                             int maxBatchSizeInBytes,
-                             int maxFetchSizeInBytes,
-                             long deleteDelayMillis) {
-        this.logSegmentBytes = logSegmentBytes;
-        this.logSegmentMinBytes = logSegmentMinBytes;
-        this.logSegmentMillis = logSegmentMillis;
-        this.retentionMaxBytes = retentionMaxBytes;
-        this.retentionMillis = retentionMillis;
-        this.maxBatchSizeInBytes = maxBatchSizeInBytes;
-        this.maxFetchSizeInBytes = maxFetchSizeInBytes;
-        this.deleteDelayMillis = deleteDelayMillis;
-    }
+    private final int internalMaxBatchSizeInBytes;
+    private final int internalMaxFetchSizeInBytes;
+    private final long internalDeleteDelayMillis;
 
     public MetadataLogConfig(AbstractConfig config) {
         this.logSegmentBytes = config.getInt(METADATA_LOG_SEGMENT_BYTES_CONFIG);
-        this.logSegmentMinBytes = config.getInt(METADATA_LOG_SEGMENT_MIN_BYTES_CONFIG);
+        this.internalSegmentBytes = config.getInt(INTERNAL_METADATA_LOG_SEGMENT_BYTES_CONFIG);
         this.logSegmentMillis = config.getLong(METADATA_LOG_SEGMENT_MILLIS_CONFIG);
         this.retentionMaxBytes = config.getLong(METADATA_MAX_RETENTION_BYTES_CONFIG);
         this.retentionMillis = config.getLong(METADATA_MAX_RETENTION_MILLIS_CONFIG);
-        this.maxBatchSizeInBytes = KafkaRaftClient.MAX_BATCH_SIZE_BYTES;
-        this.maxFetchSizeInBytes = KafkaRaftClient.MAX_FETCH_SIZE_BYTES;
-        this.deleteDelayMillis = ServerLogConfigs.LOG_DELETE_DELAY_MS_DEFAULT;
+        this.internalMaxBatchSizeInBytes = config.getInt(INTERNAL_MAX_BATCH_SIZE_IN_BYTES_CONFIG);
+        this.internalMaxFetchSizeInBytes = config.getInt(INTERNAL_MAX_FETCH_SIZE_IN_BYTES_CONFIG);
+        this.internalDeleteDelayMillis = config.getLong(INTERNAL_DELETE_DELAY_MILLIS_CONFIG);
     }
 
     public int logSegmentBytes() {
         return logSegmentBytes;
     }
-
-    public int logSegmentMinBytes() {
-        return logSegmentMinBytes;
+    
+    public Integer internalSegmentBytes() {
```

Comment:
`OptionalInt` might be nicer here.

Reply:
Since `config.getInt(INTERNAL_METADATA_LOG_SEGMENT_BYTES_CONFIG)` may return a null value, using `OptionalInt` is not suitable here.
If we want to wrap the result with `Optional`, I think `Optional<Integer>` would be more appropriate.